src/Microsoft.AspNet.Mvc.Core/ViewComponents/DefaultViewComponentInvokerProvider.cs

```diff
@@ -9,16 +9,16 @@ namespace Microsoft.AspNet.Mvc
     public class DefaultViewComponentInvokerProvider : IViewComponentInvokerProvider
     {
         private readonly IServiceProvider _serviceProvider;
-        private readonly ITypeActivator _typeActivator;
+        private readonly IViewComponentFactory _viewComponentFactory;
         private readonly IViewComponentActivator _viewComponentActivator;
```

Comment:
I think it will be too big. I got a chance to talk to @yishaigalatzer and he said we should separate it from this PR. I will follow up with you tomorrow on the exact details of whats needed here and will file a separate issue. Please let me know if you have any concerns with that.